docs/writing-benchmarks.md

```diff
@@ -0,0 +1,77 @@
+## Writing Benchmarks
+
+To get started, let's look at a simple example:
+
+```kotlin
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 1, time = 1, timeUnit = TimeUnit.SECONDS)
+@Fork(1)
+@BenchmarkTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+class ExampleBenchmark {
+
+    @Param("1", "2")
+    var param: Int = 0
+
+    private var list: List<Int> = ArrayList()
+
+    @Setup
+    fun prepare() {
+        list = List(param) { it }
+    }
+
+    @Benchmark
+    fun benchmarkMethod(): Int {
+        return list.sum()
+    }
+
+    @TearDown
+    fun cleanup() {
+        list = ArrayList()
+    }
+}
+```
+
+### Explaining the Annotations
+
+#### @State
+
+The `@State` annotation, when set to `Scope.Benchmark`, is applied to a class to represent that this class is responsible for holding the state or data for your benchmark tests. This class instance is shared among all benchmark threads, creating a consistent environment across all warmups, iterations, and measured executions. It helps in managing stateful operations in a multi-threaded context, providing an efficient way of handling state that's thread-safe and consistent across multiple runs. This ensures accurate and reliable benchmarking results, as the shared state remains the same throughout all the tests, preventing any discrepancies that could affect the outcomes. JMH provides other scope options besides `Scope.Benchmark` including `Scope.Thread`, which means each thread gets its own state instance, and `Scope.Group`, where the state is shared within a benchmark thread group.  In our snippet, the ExampleBenchmark class uses the @State(Scope.Benchmark) annotation, indicating that the state in this class is shared across all benchmark threads.
+
+#### @Setup
+
+The `@Setup` annotation is used to mark a method that sets up the necessary preconditions for your benchmark test. This method, annotated with `@Setup`, is executed before the actual benchmark methods are run. It serves as a preparatory step where you set up the environment for the benchmark, performing tasks such as generating data, establishing database connections, or preparing any other resources your benchmark requires. The key point to remember is that the `@Setup` method's execution time is not included in the final benchmark results - the timer starts only when the `@Benchmark` method begins. This makes `@Setup` an ideal place for initialization tasks that should not impact the timing results of your benchmark. By using the `@Setup` annotation, you ensure consistency across all executions of your benchmark, providing accurate and reliable results. In the provided example, the `@Setup` annotation is used to prepare a list of integers from 0 to a specified parameter value before the benchmark method is run.
```

Comment:
What levels can the `Setup` annotation have?

Reply:
Their is Level.Trial and Level.Iteration however I am assuming this is only for Kotlin/JVM and Java? 

I am getting this error when I try to use them in Kotlin/JS 
`Unresolved reference: Level`
`Too many arguments for public constructor Setup() defined in kotlinx.benchmark.Setup`

Reply:
Yes, only for Kotlin/JVM.
For other targets `Setup` and `Teardown` work on Trial level. And there is no way to adjust it.

Reply:
@qurbonzoda Thanks for clarifying. Curious as to where this ranks on our priority list in terms of adding support for adjustable levels in other targets? Other than that, the documentation has been updated to reflect the current state.

Reply:
I don't think we have enough time to implement this.

Reply:
Definitely after documentation and integration tests